airbyte-workers/src/main/java/io/airbyte/workers/internal/AirbyteMessageTracker.java

```diff
@@ -145,19 +145,28 @@ private void handleSourceEmittedRecord(final AirbyteRecordMessage recordMessage)
    * correctly.
    */
   private void handleSourceEmittedState(final AirbyteStateMessage stateMessage) {
-    updateMaxAndMeanSecondsToReceiveStateMessage(DateTime.now());
+    final LocalDateTime timeEmittedStateMessage = LocalDateTime.now();
+    stateMetricsTracker.updateMaxAndMeanSecondsToReceiveStateMessage(timeEmittedStateMessage);
     sourceOutputState.set(new State().withState(stateMessage.getData()));
-    totalSourceEmittedStateMessages.incrementAndGet();
+    stateMetricsTracker.incrementTotalSourceEmittedStateMessages();
     final int stateHash = getStateHashCode(stateMessage);
+
     try {
       if (!unreliableCommittedCounts) {
         stateDeltaTracker.addState(stateHash, streamToRunningCount);
       }
+      if (!unreliableStateTimingMetrics) {
+        stateMetricsTracker.addState(stateMessage, stateHash, timeEmittedStateMessage);
+      }
     } catch (final StateDeltaTracker.StateDeltaTrackerException e) {
       log.warn("The message tracker encountered an issue that prevents committed record counts from being reliably computed.");
       log.warn("This only impacts metadata and does not indicate a problem with actual sync data.");
       log.warn(e.getMessage(), e);
       unreliableCommittedCounts = true;
+    } catch (final StateMetricsTracker.StateMetricsTrackerException e) {
```

Comment:
this is probably confusing since I bundled it with the StateDeltaTracker, but it actually is the only error that can occur for the StateMetricsTracker. I can reword the error message too since I think it's not clear.